core/src/test/java/org/infinispan/eviction/impl/BaseEvictionFunctionalTest.java

```diff
@@ -44,6 +48,13 @@ public void testSimpleEvictionMaxEntries() throws Exception {
       }
       Thread.sleep(1000); // sleep long enough to allow the thread to wake-up
       assert CACHE_SIZE >= cache.size() : "cache size too big: " + cache.size();
+      assert CACHE_SIZE == evictionListener.getEvictedEvents() : "eviction events count should be same with case size: " + evictionListener.getEvictedEvents();
+
+      for (int i = 0; i < CACHE_SIZE; i++) {
+         cache.put("key-" + (i + 1), "value-" + (i + 1), 1, TimeUnit.MINUTES);
+      }
+      Thread.sleep(1000); // sleep long enough to allow the thread to wake-up
+      assert CACHE_SIZE >= cache.size() : "cache size too big: " + cache.size();
```

Comment:
Actually these tests are a bit out dated, eviction should happen on the same thread.  So a sleep shouldn't be required.  Although I am not sure why this is also testing expiration at the same time.